File: TP02/Series.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

struct Serie
{
    char nome[100];
    char formato[100];
    char duracao[100];
    char paisDeOrigem[100];
    char idioma[100];
    char emissora[100];
    char transmissao[100];
    int numeroDeTemporadas;
    int numeroDeEpisodios;
};

int isFim(char *s)
{
    int length = strlen(s);
    return (length == 3 && s[0] == 'F' && s[1] == 'I' && s[2] == 'M');
}

char *removeTags(char *s)
{
    char *retorno = (char *)malloc(10 * sizeof(s));
    int pos = 0;

    for (int i = 0; i < strlen(s); i++)
    {
        if (s[i] == '<')
        {
            while (s[i] != '>' && i < strlen(s))
            {
                i++;
            }
        }
        else if (i + 5 < strlen(s) && s[i] == '&' && s[i + 1] == '#' && s[i + 2] == '1' && s[i + 3] == '6' && s[i + 4] == '0' && s[i + 5] == ';')
        {
            i += 5; // Retirando "&#160;" da string
        }
        else if (i + 5 < strlen(s) && s[i] == '&' && s[i + 1] == 'n' && s[i + 2] == 'b' && s[i + 3] == 's' && s[i + 4] == 'p' && s[i + 5] == ';')
        {
            i += 5; // Retirando "&nbsp;" da string
        }
        else
        {
            if (s[i] != '\n')
            { // Retirando "\n" da string
                retorno[pos] = s[i];
                retorno[pos + 1] = '\0';
                pos++;
            }
        }
    }

    return retorno;
}

int main()
{
    int numEntrada = 0;
    char arrayStrings[1000][1000];

    do
    {
        fgets(arrayStrings[numEntrada], 1000, stdin); // Leitura da linha
        setbuf(stdin, NULL);
        arrayStrings[numEntrada][strcspn(arrayStrings[numEntrada], "\n")] =
            0; // Retira o caractere de '/n' do fim da string
    } while (isFim(arrayStrings[numEntrada++]) ==
             0); // Checa se a linha é igual a "FIM"
    numEntrada--;

    struct Serie series[numEntrada];
    for (int i = 0; i < numEntrada; i++)
    {
        char diretorio[50] = "/tmp/series/";
        strcat(diretorio, arrayStrings[i]);

        FILE *fp;
        if ((fp = fopen(diretorio, "r")) == NULL)
        {
            // Captura de erro
            printf("Erro: fopen (%s)\n", diretorio);
        }
        else
        {
            char nome[50];
            for (int j = 0; j < strlen(arrayStrings[i]) && arrayStrings[i][j] != '.';
            j++)
            {
                if (arrayStrings[i][j] == '_')
                {
                    nome[j] = ' ';
                }
                else
                {
                    nome[j] = arrayStrings[i][j];
                }
                nome[j + 1] = '\0';
            }
            strcpy(series[i].nome, nome);

            char linha[2000];
            fgets(linha, 2000, fp);

            // Realiza a leitura linha por linha até encontrar as palavras chaves
            while (strstr(linha, "Formato") == NULL)
            {
                fgets(linha, 2000, fp);
                // printf("dbg %s\n", linha);
            }
            fgets(linha, 2000, fp);
            strcpy(series[i].formato, removeTags(linha));

            while (strstr(linha, "Duração") == NULL)
            {
                fgets(linha, 2000, fp);
            }
            fgets(linha, 2000, fp);
            strcpy(series[i].duracao, removeTags(linha));

            while (strstr(linha, "País de origem") == NULL)
            {
                fgets(linha, 2000, fp);
            }
            fgets(linha, 2000, fp);
            strcpy(series[i].paisDeOrigem, removeTags(linha));

            while (strstr(linha, "Idioma original") == NULL)
            {
                fgets(linha, 2000, fp);
            }
            fgets(linha, 2000, fp);
            strcpy(series[i].idioma, removeTags(linha));

            while (strstr(linha, "Emissora de televisão original") == NULL)
            {
                fgets(linha, 2000, fp);
            }
            fgets(linha, 2000, fp);
            strcpy(series[i].emissora, removeTags(linha));

            while (strstr(linha, "Transmissão original") == NULL)
            {
                fgets(linha, 2000, fp);
            }
            fgets(linha, 2000, fp);
            strcpy(series[i].transmissao, removeTags(linha));

            while (strstr(linha, "N.º de temporadas") == NULL)
            {
                fgets(linha, 2000, fp);
            }
            fgets(linha, 2000, fp);
            char aux[50];
            strcpy(aux, removeTags(linha));
            series[i].numeroDeTemporadas = (int)strtol(aux, NULL, 10);

            while (strstr(linha, "N.º de episódios") == NULL)
            {
                fgets(linha, 2000, fp);
            }
            fgets(linha, 2000, fp);
            strcpy(aux, removeTags(linha));
            series[i].numeroDeEpisodios = (int)strtol(aux, NULL, 10);
        }
    }

    for (int i = 0; i < numEntrada; i++)
    {
        // Imprimindo resultados
        printf("%s %s %s %s %s %s %s %d %d\n", series[i].nome, series[i].formato,
            series[i].duracao, series[i].paisDeOrigem, series[i].idioma,
            series[i].emissora, series[i].transmissao,
            series[i].numeroDeTemporadas, series[i].numeroDeEpisodios);
    }
}

File: TP02/Series.java
import java.io.InputStreamReader;
import java.io.BufferedReader;
import java.io.File;
import java.io.FileInputStream;
import java.io.Reader;
import java.nio.charset.StandardCharsets;
import java.util.Scanner;

import javax.print.attribute.SetOfIntegerSyntax;

public class Series {
    
    private String nome;
    private String formato;
    private String duracao;
    private String pais;
    private String idioma;
    private String emissora;
    private String transmissao;
    private int temp;
    private int ep;


    Series() {
        nome = "";
    }
    public Series clone(){
        Series resp = new Series();
        resp.nome = this.nome;
        resp.formato = this.formato;
        resp.duracao = this.duracao;
        resp.pais = this.pais;
        resp.idioma = this.idioma;
        resp.emissora = this.emissora;
        resp.transmissao = this.transmissao;
        resp.temp = this.temp;
        resp.ep = this.ep;

        return resp;
    }
        public Series(String data[]) {
        this.nome = data[0];
        this.formato = data[1];
        this.duracao = data[2];
        this.pais = data[3];
        this.idioma = data[4];
        this.emissora = data[5];
        this.transmissao = data[6];
        this.temp = Integer.parseInt(data[7]);
        this.ep = Integer.parseInt(data[8]);
    }

    public String getnome() {
        return nome;
    }

    public String getformato() {
        return formato;
    }

    public String getduracao() {
        return duracao;
    }

    public String getpais() {
        return pais;
    }

    public String getemissora() {
        return emissora;
    }

    public String gettransmissao() {
        return transmissao;
    }

    public int gettemp() {
        return temp;
    }

    public int getep() {
        return ep;
    }


    public void setnome(String nome) {
        this.nome = nome;
    }

    public void setformato(String formato) {
        this.formato = formato;
    }

    public void setduracao(String duracao) {
        this.duracao = duracao;
    }

    
    public void setpais(String pais) {
        this.pais = pais;
    }

    public void setemissora(String emissora) {
        this.emissora = emissora;
    }

    public void settransmissao(String transmissao) {
        this.transmissao = transmissao;
    }

    public void settemp(int temp) {
        this.temp = temp;
    }

    public void setep(int ep) {
        this.ep = ep;
    }

    public String removeTags(String line) {
        String newLine = "";
        int i = 0;
        while (i < line.length()) {
            if (line.charAt(i) == '<') {
                i++;
                while (line.charAt(i) != '>')
                    i++;
            } 
            else {
                newLine += line.charAt(i);
            }
            i++;
        }
        while (newLine.contains("&") && newLine.contains(";"))
            newLine = newLine.split(";")[1];

        return newLine;
    }
    /* Lê todos os dados necessarios */
    public void ler(String nomeArq) throws Exception {
        File file = new File(nomeArq);
        BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream(file), "UTF-8"));

        String[] tmp = nomeArq.split("/");
        this.nome = tmp[tmp.length - 1];
        this.nome = this.nome.replace(".html", "").replaceAll("_", " ");

        while (!reader.readLine().contains("Formato"))
            ;
        this.formato = this.removeTags(reader.readLine());

        while (!reader.readLine().contains("Duração"))
            ;
        this.duracao = this.removeTags(reader.readLine());

        while (!reader.readLine().contains("País de origem"))
            ;
        this.pais = this.removeTags(reader.readLine());

        while (!reader.readLine().contains("Idioma original"))
            ;
        this.idioma = this.removeTags(reader.readLine());

        while (!reader.readLine().contains("Emissora de televisão original"))
            ;
        this.emissora = this.removeTags(reader.readLine());

        while (!reader.readLine().contains("Transmissão original"))
            ;
        this.transmissao = this.removeTags(reader.readLine());
        if (this.transmissao.contains("&#160;"))
            this.transmissao.replace("&#160;", "");

        while (!reader.readLine().contains("temporadas"))
            ;
        this.temp = Integer.parseInt(this.removeTags(reader.readLine()).split(" ")[0]);

        while (!reader.readLine().contains("episódios"))
            ;
        this.ep = Integer.parseInt(this.removeTags(reader.readLine()).split(" ")[0]);

        reader.close();
    }

    /* Mostra na tela todos os dados necessarios */
        public void print() {
            String out = this.nome + ' ' + this.formato + ' ' + this.duracao + ' ' + this.pais + ' '
                    + this.idioma + ' ' + this.emissora + ' ' + this.transmissao + ' '
                    + this.temp + ' ' + this.ep;
            out = out.replaceAll("  ", " ");
            System.out.println(out);
        }




    public static void main(String args[]) throws Exception {
        Scanner s = new Scanner(System.in);
        String[] seriesList = new String[200];
        int count = 0;
        while (true) {
            String line = s.nextLine();

            if (line.equals("FIM"))
                break;

            seriesList[count] = line;
            count++;
        }

        Series serie = new Series();

        for (int i = 0; i < count; i++) {
            String file = seriesList[i];
            String filePath = "/tmp/series/" + file;
            serie.ler(filePath);
            serie.print();
        }
    }

}